firebase/functions/index.js

```diff
@@ -0,0 +1,339 @@
+        questionOptionsString = " Option A: " + question.options[0] + " ,Option B: " + question.options[1] + " ,Option C: " + question.options[2] + " ,Option D: " + question.options[3];
+    } else if ( question.type==="SUBJECTIVE" ){
+        questionOptionsString = "This is a subjective type question. So no options are available for this question";
+    }
+    return {
+        points: question.points,
+        question_statement: questionStatementString,
+        options: questionOptionsString,
+        question_number: questionNum
+    } ;
+}
+
+/**
+ * Called when a new question is to be loaded into the current question context then accordingly have to set current answer
+ * Using the question number of current question and answers list from context load the current answer
+ * @param questionNum ( question number of current question )
+ * @param answerObject
+ * @returns {{answer: *}}
+ */
+function getCurrentAnswerParameter ( questionNum, answerObject ){
+    return {
+        answer:answerObject[questionNum-1]
+    };
+}
+
+/**
+ * Called when a new question is to be loaded into the context and then the question needs to be converted into a statement
+ * This function returns a conversational message to be sent to the agent.
+ * Here it normally lists all the question parameters to be told to the user
+ * @param questionObject
+ * @returns {string}
+ */
+function returnQuestionString ( questionObject ){
+    return ( "Question number " + questionObject.question_number + " for "
+        + questionObject.points + " points. " + questionObject.question_statement + ". " + questionObject.options );
+}
+
+/**
+ * Called to create user context object when starting exam intent is triggered
+ * Takes all the user context parameters as arguments of function and forms a new user context object
+ * @param user ( contains user roll number and exam id )
+ * @param exam_instance_id
+ * @param end_time ( a timestamp for end time of exam )
+ * @returns {{end_time: *, roll_number: *, exam_instance_id: *, exam_id: *}}
+ */
+function getUserContextParameter ( user, exam_instance_id, end_time ){
+
+    return {
+        roll_number: user.roll_number,
+        exam_id: user.exam_id,
+        exam_instance_id: exam_instance_id,
+        end_time: end_time
+    }
+
+}
+
+/**
+ * Handles the exam.start.submit_request intent
+ * contexts set -> user ( containing user data to be used when submitting exam )
+ *                 exam ( exam object )
+ *                 answers ( answers object )
+ *                 current_question ( statement, options, points, question number )
+ *                 current_answer ( answer )
+ * contexts deleted to avoid confusion and avoiding editing by user:
+ *                 roll_number, exam_instance_id, exam_start_request
+ * If error is encountered then send appropriate message in response
+ */
+app.intent( 'exam.start.submit_request', async conv => {
+
+    try {
+        const userData = await getUserData(conv);
```

Comment:
Adding everything in try/catch is not a good idea.

Reply:
Updated.